pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -178,7 +180,13 @@ func NewController(client kube.Client, meshWatcher mesh.Holder,
 	c.queue = controllers.NewQueue("ingress",
 		controllers.WithReconciler(c.onEvent),
 		controllers.WithMaxAttempts(5))
-	c.ingressInformer.AddEventHandler(controllers.ObjectHandler(c.queue.AddObject))
+	c.ingressInformer.AddEventHandler(
+		controllers.FilteredObjectHandler(c.queue.AddObject, func(o controllers.Object) bool {
```

Comment:
shouldn't we be filtering the informer itself, not the handler? We may be reading Ingresses outside of the handler events

Reply:
I see. I was looking at NewFilteredSharedIndexInformer and thought it was in core client-go doing it under the hood. Looks like not.

Can we re-use NewFilteredSharedIndexInformer though? It will do the same thing but avoid re-implementing the same (somewhat complex) logic everywhere?

Probably can move it to `pkg/kube` or similar

Reply:
That looks good, i will do that